spec/re2_spec.rb

```diff
@@ -15,29 +15,68 @@
     it "does not perform replacements in-place" do
       name = "Robert"
       replacement = RE2.Replace(name, "R", "Cr")
-      expect(name).to_not equal(replacement)
+
+      expect(name).not_to equal(replacement)
     end
 
     it "supports passing an RE2::Regexp as the pattern" do
       re = RE2::Regexp.new('wo{2}')
+
       expect(RE2.Replace("woo", re, "miaow")).to eq("miaow")
     end
 
     it "respects any passed RE2::Regexp's flags" do
       re = RE2::Regexp.new('gOOD MORNING', :case_sensitive => false)
+
       expect(RE2.Replace("Good morning", re, "hi")).to eq("hi")
     end
 
-    if String.method_defined?(:encoding)
-      it "preserves the original string's encoding" do
-        original = "Foo"
-        replacement = RE2.Replace(original, "oo", "ah")
-        expect(original.encoding).to eq(replacement.encoding)
-      end
+    it "supports passing something that can be coerced to a String as input" do
+      expect(RE2.Replace(StringLike.new("woo"), "oo", "ah")).to eq("wah")
+    end
+
+    it "supports passing something that can be coerced to a String as a pattern" do
+      expect(RE2.Replace("woo", StringLike.new("oo"), "ah")).to eq("wah")
+    end
+
+    it "supports passing something that can be coerced to a String as a replacement" do
+      expect(RE2.Replace("woo", "oo", StringLike.new("ah"))).to eq("wah")
+    end
+
+    it "returns UTF-8 strings if the pattern is UTF-8" do
+      original = "Foo".encode("Shift_JIS")
+      replacement = RE2.Replace(original, "oo", "ah")
+
+      expect(replacement.encoding).to eq(Encoding::UTF_8)
+    end
+
+    it "returns ISO-8859-1 strings if the pattern is not UTF-8" do
+      original = "Foo".encode("Shift_JIS")
+      replacement = RE2.Replace(original, RE2("oo", :utf8 => false), "ah")
+
+      expect(replacement.encoding).to eq(Encoding::ISO_8859_1)
+    end
+
+    it "returns strings in the encoding of the given String pattern" do
```

Comment:
It’s worth noting this probably isn’t the right behaviour: the pattern will be compiled into an `RE2` but the input will be interpreted as UTF-8 by default so any result will also be UTF-8. (Not to mention this will probably result in undefined behaviour if passing non-UTF-8 data.)